utils/lit/lit/formats/taef.py

```diff
@@ -80,15 +82,24 @@ def getTestsInDirectory(self, testSuite, path_in_suite,
                 litConfig, localConfig):
             yield test
 
+
     def execute(self, test, litConfig):
         test_dll = test.getFilePath()
 
         testPath,testName = os.path.split(test.getSourcePath())
 
+        select_filter = str.format("@Name='{}'", testName)
+
+        # FIXME: enable mix filter.
+        #if self.select_filter != "":
+        #    select_filter = str.format("{} AND {}", select_filter, self.select_filter)
+
+        select_filter = str.format('/select:{}', select_filter)
+
         cmd = [self.te, test_dll, '/inproc',
+                select_filter,
                 '/miniDumpOnCrash', '/unicodeOutput:false',
-                str.format('/outputFolder:{}', self.test_path),
-                str.format('/name:{}', testName)]
+                str.format('/outputFolder:{}', self.test_path)]
```

Comment:
I think we need to change this whole function for TAEF execution, because apparently, TAEF custom-processes the command line instead of letting the CRT process it first.  This method of passing arguments will cause them to be quoted if necessary by subprocess.Popen, but the quotes won't be where TAEF wants them, so it won't recognize the arguments properly at all.

Instead, this needs to construct the full command line string and use `shell=True` when invoking `subprocess.Popen()`.  Each injected value that could contain characters that need quoting (path names, `select_filter`) need to be quoted just around the value.  Like:

```py
def execute(self, test, litConfig):

    test_dll = test.getFilePath()

    testPath,testName = os.path.split(test.getSourcePath())

    select_filter = str.format("@Name='{}'", testName)

    if self.select_filter != "":
        select_filter = str.format("{} AND {}", select_filter, self.select_filter)

    select_filter = str.format('/select:"{}"', select_filter)

    cmd = [self.te, test_dll, '/inproc',
        select_filter,
        '/miniDumpOnCrash', '/unicodeOutput:false',
        str.format('/outputFolder:"{}"', self.test_path)]

    cmd.extend(self.extra_params)

    if litConfig.useValgrind:
        cmd = litConfig.valgrindArgs + cmd
    if litConfig.noExecute:
        return lit.Test.PASS, ''

    # TAEF must be run with custom command line string and shell=True
    # because of the way it manually processes quoted arguments in a
    # non-standard way.
    command = ' '.join(cmd)

    print(command)

    p = subprocess.Popen(command, cwd=cwd, shell=True,
                         stdin=subprocess.PIPE,
                         stdout=subprocess.PIPE,
                         stderr=subprocess.PIPE,
                         env=test.config.environment,
                         close_fds=lit.util.kUseCloseFDs)
    out,err = p.communicate()
    exitCode = p.wait()

    # Detect Ctrl-C in subprocess.
    if exitCode == -signal.SIGINT:
        raise KeyboardInterrupt

    # Ensure the resulting output is always of string type.
    out = convert_string(out)
    err = convert_string(err)

    ...
```

Also need to quote a few other variables set in `lit.cfg` files elsewhere:
`param_hlsl_data_dir = str.format('/p:HlslDataDir="{}"', hlsl_data_dir)`
`extra_params.append(str.format('/p:Adapter="{}"', adapter))`

Reply:
Thanks a lot for work this out.